src/v2/controllers/FormController.js

```diff
@@ -156,7 +161,22 @@ export default Controller.extend({
         .then(this.handleIdxSuccess.bind(this))
```

Comment:
I wonder if you shall suppose to change here instead of adding another `.then` ?
```
.then(function () {
  if (action === 'cancel' AND useInteractionFlow) {
    this.options.appState.trigger('interactionCanceled');
  } else {
     this.handleIdxSuccess();
  }
});
```

Reply:
Done